utils/docker/docker-utils

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+
+"""
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+ -------------------------------------------------------------------------
+ This script contains helper operations to use Docker for SwitPM development
+ and testing.
+"""
+
+import argparse
+import urllib2
+import os
+import subprocess
+import yaml
+
+def call(args):
+    """Prints and executes a command."""
+    print " ".join(args)
+    return subprocess.call(args)
+
+def docker_run(command_args):
+    """Runs a command in the container."""
+    call([
+        "docker",
+        "run",
+        "-it",
+        "--security-opt", "seccomp=unconfined",
+        "-v", os.getcwd() + ":/llbuild",
+        "-w", "/llbuild",
+        "--rm",
+        "llbuild-docker-1604"
+    ] + command_args)
+
+def build(_):
+    """Builds a docker image with the latest snapshot."""
+
+    # Get latest snapshot info.
+    latest_build_url = "https://swift.org/builds/development/ubuntu1604/latest-build.yml"
+    latest_build_data = yaml.load(urllib2.urlopen(latest_build_url).read())
+    snapshot_filename = latest_build_data["download"]
+    # FIXME: We shouldn"t need to do this, it should be available in the API.
+    snapshot_name = snapshot_filename.replace("-ubuntu16.04.tar.gz", "")
+    base_url = latest_build_url.rsplit("/", 1)[0]
+    latest_snapshot_url = base_url + "/" + snapshot_name + "/" + snapshot_filename
+    docker_dir = os.path.dirname(os.path.realpath(__file__))
+
+    # Download latest snapshot (if necessary).
+    if snapshot_filename in os.listdir(docker_dir):
+        print "Not downloading: " + latest_snapshot_url
+    else:
+        # FIXME: We should remove old tarballs if we have newer ones.
+        result = call([
+            "curl",
+            "-o",
+            docker_dir + "/" + snapshot_filename,
+            latest_snapshot_url
+        ])
+
+        if result != 0:
+            print "Unable to install package: " + latest_snapshot_url
+            exit(1)
+
+    # Create docker image.
+    call([
+        "docker",
+        "build",
+        "-t", "llbuild-docker-1604",
+        "--build-arg", "SNAPSHOT=%s" % snapshot_filename,
+        docker_dir
+    ])
+
+def run(args):
+    """Runs an executable in the container."""
+    docker_run([args.executable] + args.arguments)
+
+def bootstrap(args):
```

Comment:
We should remove this, right?

Reply:
Yep, sorry. I removed some but missed those.

Reply:
Removed.